### PR TITLE
SE-1909: Added a CA example for dividend, stock split, and name change

### PR DESCRIPTION
**Pull request by Tomwp**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Added a corporate action example covering dividends, stock splits, and name changes. Tests include checks before and after each event as well as between ex-dates and payment dates for stock splits and dividends

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs [hunk 1]
```diff
@@ -33,5 +102,349 @@ public void List_Corporate_Actions_For_One_Day()
...
+
+            //
+            // fetch holdings pre ex-dividend date
+            //
+            var holdingsResultPreExDate = _transactionPortfoliosApi.GetHoldings(TestDataUtilities.TutorialScope, portfolioCode, effectiveAt: announcementDate);
+            var holdingsPreExDate = holdingsResultPreExDate.Values.OrderBy(h => h.InstrumentUid).ToList();
+
+            //  check cash balance hasn't changed
+            Assert.That(holdingsPreExDate[0].InstrumentUid, Is.EqualTo(currencyLuid));
+            Assert.That(holdingsPreExDate[0].Units, Is.EqualTo(2960000));
```
Comment (cscialino) [inline sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs:310]: I wonder whether we need the cash balance check for a stock split?

Comment (cscialino) [inline sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs:310]: I wonder whether we need the cash balance check for a stock split or does it add noise? If we want to check the impact is only on the stock units, you may want to assert the number of holdings hasn't changed as a result.

File: sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs [hunk 2]
```diff
@@ -8,9 +13,73 @@
 namespace Lusid.Sdk.Tests.Tutorials.MarketData
 {
     [TestFixture]
-    public class CorporateActions: TutorialBase
+    public class CorporateActions
```
[the remark this answers is not in the document]
Comment (Tomwp) [reply sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs]: Added this back in for consistency with other test classes.

File: sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs [hunk 3]
```diff
@@ -8,9 +13,73 @@
...
+                if (errorObj.RootElement.GetProperty("code").ToString() == "391")
+                    return;
+                else
+                    throw;
+            }
+        }
+        
         [LusidFeature("F12-6")]
         [Test]
         public void List_Corporate_Action_Sources()
```
[the remark this answers is not in the document]
Comment (Tomwp) [reply sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs:77]: Agree however we have this convention in other test cases. I'd prefer to create a separate ticket to address this across the board.

File: sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs (hunk 1 above)
Comment (Tomwp) [reply sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs]: Agreed - removed.

Comment (Tomwp) [reply sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs]: Agreed - removed.

File: sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs [hunk 4]
```diff
@@ -33,5 +102,349 @@ public void List_Corporate_Actions_For_One_Day()
...
+        [Test]
+        public void Process_Name_Change_By_Transitions()
+        {
+            var uuid = Guid.NewGuid().ToString();
+
+            var portfolioCode = $"id-{uuid}";
+            var transactions = new List<TransactionRequest>();
+
+            var originalInstrument = (Figi: "BBG000C6K6G9", Name: "VODAFONE GROUP PLC");
+            var newInstrument = (Figi: "BB5555555555", Name: "VODAFONE INCORPORATED");
```
[the remark this answers is not in the document]
Comment (Tomwp) [reply sdk/Lusid.Sdk.Tests/tutorials/MarketData/CorporateActions.cs:338]: Added teardown logic.

**Pull request merged**